src/core/reference/include/openvino/reference/avg_pool.hpp

```diff
@@ -68,13 +67,13 @@ void avg_pool_backprop(const T* delta,
         }
         std::iota(begin(source_window_transform_source_axis_order), end(source_window_transform_source_axis_order), 0);
 
-        CoordinateTransform source_window_transform(out_shape,
-                                                    source_window_transform_start,
-                                                    source_window_transform_end,
-                                                    source_window_transform_source_strides,
-                                                    source_window_transform_source_axis_order,
-                                                    source_window_transform_padding_below,
-                                                    source_window_transform_padding_above);
+        ngraph::CoordinateTransform source_window_transform(out_shape,
```

Comment:
If I remember this class was deprecated by ngraph team and recommendation is to us `CoordinateTransformBase` can we remove `NGRAPH_SUPPRESS_DEPRECATED_START` and use alternatives instead of deprecated classes?

Reply:
This is addressed by separate story CVS-118401 and will be started shortly.

Reply:
I'll move `CoordinateTransform...`s into `ov::` namespace in this PR. But dropping this deprecated will be in separate PR.

Reply:
If I remember `CoordinateTransform` had an issue with the performance....
In this case I think we need to take a look and maybe use `CoordinateTransformBase` and do not introduce deprecated class to `ov` namespace

Reply:
I agree, it needs to be refactored. To not mix the concerns I'll handle that in separate PR.

Reply:
@ilyachur 
Can we postpone the `ngraph::CoordinateTransform` replace to `CoordinateTransformBase` in separate tasks?
It looks like is this can be changed when refactoring and dropping `ngraph` namespace from op implementation and rest can be done in mentioned Task CVS-118401

This PR has a lot of files changed and changing CoordinateTransform can impact on test (we can have some failures), I think would be easier to manage errors if just replace deprecated class want be enough.

Reply:
Yes of course it can be done in next PRs